### PR TITLE
Fix image extension tests

### PR DESCRIPTION
**Pull request by fmassa**
Follow-up from https://github.com/pytorch/vision/pull/2379#discussion_r448862180

Thanks to @peterjc123 , we found out that there was a bug in the tests that were preventing them to run.

### CODE REVIEW

**Comment by fmassa** [comment]
@andfoy Seems like Windows is not able to find libpng. I'm going to be reverting your PR again, can you integrate the changes in this branch and then try fixing the issues with Windows?

**Comment by peterjc123** [comment]
Hi @andfoy, I've made a few suggestions in https://github.com/pytorch/vision/pull/2379#pullrequestreview-441751366. Hopefully that may help you to get libpng to be found before installation.

**Comment by andfoy** [comment]
Thanks @peterjc123! Let me apply those suggestions here and hopefully it will work as expected

File: setup.py [hunk 1]
```diff
@@ -171,6 +230,75 @@ def get_extensions():
...
+                      'disabling PNG support')
+                png_found = False
+        else:
+            # Windows
+            png_lib = os.path.join(
+                os.path.dirname(os.path.dirname(pngfix)), 'lib')
+            png_include = os.path.join(os.path.dirname(
+                os.path.dirname(pngfix)), 'include', 'libpng16')
+            image_library += [png_lib]
+            image_include += [png_include]
```
Comment (peterjc123) [inline setup.py:286]: What about `image_link_flags`?

File: torchvision/io/image.py [hunk 2]
```diff
@@ -37,7 +37,8 @@ def decode_png(input):
     Returns:
         output (Tensor[image_width, image_height, 3])
     """
-    if not isinstance(input, torch.Tensor) or input.numel() == 0 or input.ndim != 1:
+    if not (isinstance(input, torch.Tensor) or
+            input.numel() == 0 or input.ndim != 1):
```
Comment (fmassa) [inline torchvision/io/image.py:41]: This introduced a bug in the code, as the parenthesis is now negating the whole expression, and not only the `isinstance`

File: packaging/torchvision/meta.yaml [hunk 3]
```diff
@@ -8,6 +8,7 @@ source:
 requirements:
   build:
     - {{ compiler('c') }} # [win]
+    - libpng
```
Comment (peterjc123) [inline packaging/torchvision/meta.yaml:11]: Are you sure this is only a build-time dependency? Have you copied the libs for the conda package?

**Comment by andfoy** [comment]
https://github.com/pytorch/pytorch/pull/40928 is passing, so that means that the documentation build was fixed as well

**Comment by peterjc123** [comment]
@fmassa You can merge it if you like.

**Comment by fmassa** [comment]
Thanks @andfoy and @peterjc123 !

**Pull request merged**